docs/DeveloperGuide.md

```diff
@@ -1,11 +1,216 @@
+
+
+- **Step3**
+
+  `AddRecipeCommand` will add newly created `Recipe` into `RecipeList`. Then, the recipe will be added into an
+  `ArrayList` inside `RecipeList`.
+
+
+- **Step4**
+
+  `AddRecipeCommand` will call `Ui` class to print out the title of the recently added recipe.
+
+<img src="images/AddNewRecipeSequenceDiagram.png" width="963" />
+
+### View Ingredients feature
+The view ingredient feature is facilitated by the `ViewIngredientCommand` class. Users can input
+"view i" to trigger this command. Users will then be able to see all ingredients stored.
+Example: 
+
+`1. bread: 2pcs`
+
+`2. apple: 500g`
+
+`...`
+
+- **Step 1**
+
+  Input will be sent from the main `EssenMakanan` class to the `Parser` to identify the command type.
+
+
+- **Step 2**
+  
+  A new `ViewIngredientCommand` object will be created and will be sent back to main 
+
+
+- **Step 3**
+  
+  `commandObject#executeCommand()` will be called which in turn calls `Ui#printAllIngredients()`
+
+
+- **Step 4**
+  
+  Finally `Ui#printAllIngredients()` will call `IngredientList#listIngredients()` to print all the ingredients
+  to standard output
+
+
+<img src="images/ViewAllIngredientSequenceDiagram.png" width="963" />
+
+### View Recipes feature
+The view recipes feature is facilitated by the `ViewRecipeCommand` class. Users can input
+"view r" to trigger this command
+
+- **Step 1**
+
+  Input will be sent from the main `EssenMakanan` class to the `Parser` to identify the command type.
+
+- **Step 2**
+
+  A new `ViewRecipeCommand` object will be created and will be sent back to main
+
+- **Step 3**
+
+  `commandObject#executeCommand()` will be called which in turn calls `Ui#printAllRecipes()`
+
+- **Step 4**
+
+  Finally, `RecipeList#listRecipeTitles()` will be called to print all the ingredients
+  to standard output
+
+<img src="images/ViewAllRecipeSequenceDiagram.png" width="963" />
+
+### Add Ingredient feature
+
+The add Ingredient feature is used by a `AddIngredientCommand` class.
+Multiple ingredients can be added at the same time using the syntax
+`add i/NAME,QUANTITY,UNIT i/INGREDIENT2,.. i/INGREDIENT3...`
+
+By calling `executeCommand` on the class, the steps will
+be executed as follows:
+- **Step1**
+
+  `AddIngredientCommand` will use the "split" method of `String` to get an array of the descriptions of ingredients and iterate all the elements in this array
+
+
+- **Step2**
+
+  `AddIngredientCommand` will get a new `Ingredient` by invoking the method "parseIngredient" of `IngredientParser` for each element of the obtained array
+
+
+- **Step3**
+
+  `AddIngredientCommand` will add this `Ingredient` into `IngredientList`. Then, the ingredient will be added into an
+  `ArrayList` inside `IngredientList`.
+
+
+- **Step4**
+
+  `AddIngredientCommand` will call `Ui` class to print out the name of the recently added ingredient.
+
+<img src="images/AddNewIngredientSequenceDiagram.png" width="963" />
```

Comment:
Perhaps the sequence diagrams for various commands could be combined, instead of having separate diagrams for each command, such as with alt blocks.